plugins.md

```diff
@@ -0,0 +1,225 @@
+    express: path.join(__dirname, 'plugins/express.js')
+  }
+})
+
+At this time, the path to the plugin may either be a module name or an absolute
+path to the plugin.
+
+## What the API provides
+
+The `api` object, in short, provides functions that facilitate the following:
+
+- Creating trace spans and add labels to them.
+- Getting information about how the trace agent was configured in the
+  current application.
+- Parsing and serializing trace contexts for the sake of propagating them over
+  the network.
+- Binding callbacks and event emitters with a CLS namespace in order to
+  propagate trace contexts across asynchronous boundaries.
+
+In addition to the above, the `api` object also provides a number of well-known
+label keys and constants through `api.labels` and `api.constants` respectively.
+
+### Trace Spans
+
+These functions provide the capability to create trace spans, add labels to
+them, and close them. `transaction` and `childSpan` are instances of
+`Transaction` and `ChildSpan`, respectively.
+
+#### `api.createTransaction(options)`
+* `options`: [`TraceOptions`](trace-span-options)
+* Returns `Transaction`
+
+Creates and returns a new `Transaction` object. A `Transaction` object
+represents a root-level trace span, and exposes functions for adding labels,
+ending the span, and serializing the current trace context. Note that the
+underlying root span is started when this function is called.
+
+This function consults the trace agent's tracing policy to determine whether a
+trace span should actually be started or not. If the tracing policy prohibits
+a new span to be created under the current circumstances, this function will
+return `null` instead.
+
+#### `api.getTransaction()`
+* Returns `Transaction`
+
+Returns a Transaction object that corresponds to a root span started earlier
+in the same context, or `null` if one doesn't exist.
+
+#### `api.runInRootSpan(options, fn)`
+* `options`: [`TraceOptions`](trace-span-options)
+* `fn`: `function(Transaction): any`
+* Returns `any`
+
+Runs the given function in a root span corresponding to an incoming request,
+passing it the result of calling `api.createTransaction(options)`. The provided
+function should accept a nullable `Transaction` object. If `null` is provided,
+the function should proceed as if nothing is being traced.
+
+#### `transaction.runInChildSpan(options, fn)`
+* `options`: [`TraceOptions`](trace-span-options)
+* `fn`: `function(ChildSpan): any`
+* Returns `any`
+
+Runs the given function in a child span corresponding to an incoming request.
+The provided function should accept a `ChildSpan` object, which represents a
+child span.
+
+#### `transaction.addLabel(key, value) | childSpan.addLabel(key, value)`
+* `key`: `string`
+* `value`: `any`
+
+Add a label to the span associated with the calling object.
+
+#### `transaction.endSpan() | childSpan.endSpan()`
+
+Ends the span associated with the calling object. This function should only be
+called once.
```

Comment:
It's silently ignored right now. I would like to add an error message if it is, though.